File: cockpit/web/src/api/fleet.ts
import { get } from "./client";

/**
 * Befunde über die Connector-Flotte (ADR-0021 D6).
 *
 * Nichts davon ist gespeichert: die Konsole rechnet bei jeder Anfrage. Ein
 * gespeicherter Befund müsste quittiert und aufgeräumt werden — und wäre
 * spätestens dann falsch, wenn sich der Agent wieder gemeldet hat.
 */
export type Severity = "warning" | "critical";

export interface FleetFinding {
  /** Maschinenlesbar; die Ansicht gruppiert danach. */
  kind: string;
  severity: Severity;
  tenant_slug: string;
  agent_name: string | null;
  /** Für Menschen, mit dem nächsten Schritt darin. */
  detail: string;
}

export interface FleetReport {
  findings: FleetFinding[];
  /**
   * Derselbe Wert, den `cockpit_api.cli.fleet_check` als Exit-Code liefert:
   * 0 in Ordnung, 1 Warnung, 2 kritisch. Er kommt vom Server, damit die
   * Ansicht den Schweregrad nicht anders rechnet als die Überwachung.
   */
  worst: number;
}

export function getFleetReport(): Promise<FleetReport> {
  return get("/api/fleet/findings");
}

File: cockpit/api/cockpit_api/schemas/fleet.py
"""Schemata für die Flotten-Befunde (ADR-0021 D6)."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict

from cockpit_api.services.fleet import Severity


class FleetFindingOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    #: Maschinenlesbar — die Oberfläche gruppiert danach, und ein
    #: Überwachungssystem kann einzelne Arten ausblenden.
    kind: str
    severity: Severity
    tenant_slug: str
    agent_name: str | None
    #: Für Menschen, mit dem nächsten Schritt darin. Ein Befund ohne „und was
    #: nun“ ist eine Zeile, die man wegklickt.
    detail: str


class FleetReportOut(BaseModel):
    findings: list[FleetFindingOut]
    #: Derselbe Wert, den das CLI als Exit-Code zurückgibt: 0 in Ordnung,
    #: 1 Warnung, 2 kritisch. Damit muss die Oberfläche den Schweregrad nicht
    #: selbst aus der Liste rechnen — und rechnet ihn nicht anders als das CLI.
    worst: int


__all__ = ["FleetFindingOut", "FleetReportOut"]
